platemate/food/views.py

```diff
@@ -273,6 +303,39 @@ class SubmissionForm(forms.Form):
 }
 
 @transaction.atomic
+@csrf_exempt
+def api_photo_upload(request):
+    try:
+        if('HTTP_X_API_KEY' in request.META and request.META['HTTP_X_API_KEY'] == settings.API_KEY):
+            now = datetime.now()
+            time_string = now.isoformat()
+            photo = request.FILES['upload']
+            random.seed()
+            photo_name = str(random.randint(0,1000000)) + "_" + time_string + "_" + photo.name
+            static_sub_dir = 'api/photos'
+            saved_photo_url = process_photo_and_get_url(photo, static_sub_dir, photo_name)
+            p = Photo.factory(photo_url = saved_photo_url)
+            u = create_or_get_api_user()
+
+            s = Submission(
+                photo = p,
+                meal = 'B',
+                date = now,
+                user = u,
+                submitted = now,
+                manual = False
+            )
+            s.save()
+
+            data = dict({"photo_id" : p.id})
+
+            return JsonResponse(data)
+        else:
+            return HttpResponse('Unauthorized - Missing or invalid api key, please try again.', status=401)
```

Comment:
Can we make this a guard clause at the top? I prefer that pattern instead of a super long `if` block b/c it keeps things less indented and reduces anxiety :)